### DISCUSSION TITLE
'Find Stores' using postcode/ZIP code not working in default Store Locator

### DISCUSSION THREAD

**Discussion by alyap17** [SalesforceCommerceCloud/sfcc-ci#522] [Team Posts]
Hi all, has anyone experienced issues with searching stores by postcode/ZIP code using the default Store Locator? It has been linked to a valid Google Maps API key but it's not returning any results. 'Use My Location', on the other hand, is working fine. Thanks in advance

**Comment by jsampaioTech** [comment]
Hi Alyap,

To see results you need more 2 steps:

1. Configure your stores in BM with latitude and longitude. (Merchant
Tools/Online Marketing/Stores -> Address tab)
2. Import the geolocations for each country in Store Locator Data
(Administration/Global Preferences/Store Locator Data), by default, exists
to US and Germany.

If exists some store inside the radius that you select you will see the
results.

This doc can give you more info:
https://documentation.b2c.commercecloud.salesforce.com/DOC2/index.jsp

**Comment by bolopaul** [comment]
Hi jsampaioTech and alyap.
For work function searching stores by postcode/ZIP there is using standard **dw/catalog/StoreMgr** and function **searchStoresByPostalCode** which needs for work country code. If you look **cartridge\scripts\helpers\storeHelpers.js** the only way get it is your geolocation so it uses only your country, or if it is some problems with geolocation it takes it from current locale (**modules\server\request.js**). 
If you need work with different countries and the ability to choose a country you need additional customization.

**Comment by jsampaioTech** [comment]
Hi Pavlo,

You are right, the SFRA gets the geolocation of the request. If the idea is
to check by a country, It's necessary additional customization in the
storeHelpers.js.

**Comment by taras-pylypiv** [comment]
Hello, @alyap17
Be sure to check if Geolocations are populated for country you are trying to search in using postal code/ZIP

Administration >  Global Preferences >  Store Locator Data

Documentation:
https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/Promotions/ImportingandExportingGeolocations.html
https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/Promotions/Storegeolocationdata.html

**Comment by dimitarkdimitrov** [comment]
Guys, STOP spamming the whole group!

**Comment by alyap17** [comment]
Hi guys thanks for your responses. I've tried to search for a US based Postal Code (i.e. 06605) in the default RefArch site using a store that has the same postal code. Unfortunately, no results are coming up. Seems like the geolocation is not the issue here but rather how the postal code is searched?

Apologies to those who are getting spammed by this discussion. Feel free to unsubscribe from this discussion on the top right corner of this question. Thanks

**Comment by jimlynch-sfcc** [comment]
Hi @alyap17 : I'd suggest reporting this issue to the SFRA repo specifically to make sure this gets the correct attention, or to Commerce Cloud Support. https://github.com/SalesforceCommerceCloud/storefront-reference-architecture/issues

That said, I am unable to reproduce this in our testing environment using the postal code you provided. However, filing an issue in one of those areas will help us work through the specifics of your issue and determine if there's a bug. Thanks!

**Comment by alyap17** [comment]
sure will do thanks @jimlynch-sfcc

**Comment by aboyall-zz** [comment]
Ok cool.

How would I get it to work for Australia?

--
Alex Boyall
Lead Solution Engineer - ANZ
Commerce Cloud (B2C & B2B)
+61 415 413 019

On 6 March 2019 at 2:22 pm, alyap17 <notifications@github.com> wrote:

sure will do thanks @jimlynch-sfcc <https://github.com/jimlynch-sfcc>

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/9/comments/8>,
or mute the thread

**Comment by aboyall-zz** [comment]
Sorry all, didn't realise I had hit 'reply' to everyone.

Please ignore my below request, I'm also sorted now :)

--
Alex Boyall
Lead Solution Engineer - ANZ
Commerce Cloud (B2C & B2B)
+61 415 413 019

On 6 March 2019 at 3:15 pm, Alexander Boyall <aboyall@salesforce.com> wrote:

Ok cool.

How would I get it to work for Australia?

--
Alex Boyall
Lead Solution Engineer - ANZ
Commerce Cloud (B2C & B2B)
+61 415 413 019

On 6 March 2019 at 2:22 pm, alyap17 <notifications@github.com> wrote:

sure will do thanks @jimlynch-sfcc <https://github.com/jimlynch-sfcc>

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/9/comments/8>,
or mute the thread
